collections/union.ts

```diff
@@ -0,0 +1,25 @@
+// Copyright 2018-2021 the Deno authors. All rights reserved. MIT license.
+
+/**
+ * Returns all distinct elements that appear in either of the given arrays
+ *
+ * Example:
+ *
+ * ```typescript
+ * const soupIngredients = [ 'Pepper', 'Carrots', 'Leek' ]
+ * const saladIngredients = [ 'Carrots', 'Radicchio', 'Pepper' ]
+ * const shoppingList = union(soupIngredients, saladIngredients)
+ *
+ * console.assert(shoppingList === [ 'Pepper', 'Carrots', 'Leek', 'Radicchio' ])
+ * ```
+ */
+export function union<T>(a: Array<T>, b: Array<T>): Array<T> {
+  const s = new Set<T>();
+  while (a.length > 0) {
+    s.add(a.shift() as T);
```

Comment:
Why is the input mutated here?

Reply:
Right, this shouldn't change the input.

Reply:
Brain fart on my end, I'll update the implementation and add a test that checks the original arrays are intact

Reply:
We should add that test to all functions probably.

Reply:
> Brain fart on my end, I'll update the implementation and add a test that checks the original arrays are intact

If you are not shifting, this looks like a good case for a simple forEach call :-)

Reply:
> This looks like a good case for a simple forEach call :-)

I updated to a `for...of`